frontend/src/metabase/dashboard/components/DashboardGrid.tsx

```diff
@@ -221,48 +246,70 @@ class DashboardGridInner extends Component<
     }
   }
 
-  UNSAFE_componentWillReceiveProps(nextProps: DashboardGridProps) {
+  static getDerivedStateFromProps(
+    nextProps: DashboardGridProps,
+    state: DashboardGridState,
+  ) {
     const { dashboard, dashcardData, isEditing, selectedTabId } = nextProps;
+    const lastProps = state._lastProps;
+
+    if (
+      !lastProps ||
+      !_.isEqual(dashboard.dashcards, lastProps.dashboard.dashcards) ||
+      !_.isEqual(dashcardData, lastProps.dashcardData) ||
+      isEditing !== lastProps.isEditing ||
+      selectedTabId !== lastProps.selectedTabId
+    ) {
+      const visibleCardIds = !isEditing
+        ? getVisibleCardIds(
+            dashboard.dashcards,
+            dashcardData,
+            state.visibleCardIds,
+          )
+        : new Set(dashboard.dashcards.map(card => card.id));
+
+      const visibleCards = getVisibleCards(
+        dashboard.dashcards,
+        visibleCardIds,
+        isEditing,
+        selectedTabId,
+      );
 
-    const visibleCardIds = !isEditing
-      ? getVisibleCardIds(
-          dashboard.dashcards,
-          dashcardData,
-          this.state.visibleCardIds,
-        )
-      : new Set(dashboard.dashcards.map(card => card.id));
-
-    const cards = this.getVisibleCards(
-      dashboard.dashcards,
-      visibleCardIds,
-      isEditing,
-      selectedTabId,
-    );
+      const lastVisibleCards = lastProps?.dashboard?.dashcards
+        ? getVisibleCards(
+            lastProps.dashboard.dashcards,
+            state.visibleCardIds,
+            lastProps.isEditing,
+            lastProps.selectedTabId,
+          )
+        : [];
+
+      const hasVisibleDashcardsChanged = _.isEqual(
```

Comment:
this looks wrong, if they're equal they're not changed. You probably need to add `!` to the value and remove it from where this var is used, if you wanna keep this name.

Reply:
Good eye! It is indeed wrong, let me update it

Reply:
fixed in https://github.com/metabase/metabase/pull/53213/commits/7c9a132ecdab7b0f5654466438d5c396a2062c18